ext/windows/facter.bat

```diff
@@ -0,0 +1,8 @@
+@echo off
+SETLOCAL
+if exist "%~dp0environment.bat" (
```

Comment:
See comments on https://github.com/puppetlabs/hiera/pull/340

The interesting part about facter is that these files aren't installed by the system, they have to be purposefully copied over. I figured that since they're in the public facter repo, theoretically they should be able to be used on their own. So, if someone just installs this on their system without environment.bat, it should still work. The other batch scripts assume ruby is in the working directory, so I kept that assumption here if environment.bat doesn't exist.